### PR TITLE
Translate copy-i18n-tags to JS

### PR DESCRIPTION
**Pull request by jmif**
Key Changes:

- Translated copy-i18n-tags to vanilla JS and added comments to understand steps
- Added prototype CircleCI script to automatically run script once a week and open PR for team review (testing this out)

TODO:

- Move propagate-i18n-tags to copy-i18n-tags once approved (keeping for now to quickly compare)

### CODE REVIEW

File: scripts/propagate-i18n-tags.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+                // translation is not marked as changed until it is uncommented in a locale file.  Once it is
+                // uncommented in a translation file, the translation is considered active and changes should be
+                // tracked
+                comment = comment.replace(CHANGE_PATTERN, '');
+            } else {
+                const tagIsMarkedChangeRegex = new RegExp(`^\\s+"?${escapeRegexp(enTagName)}"?:\\s".*"\\s*` +
+                    `#\\s*${escapeRegexp(CHANGE_MARKER)}\\s*$`, 'm');
+
+                // If locale file has tag marked as change and comment is not already marked change,
+                // add change to comment
```
Comment (shubhi1092) [inline scripts/propagate-i18n-tags.js]: We can probably add more context to this comment, it looks a bit confusing.
Something like: `if en file has tag marked as change and this locale file doesnt, then add change comment to local file`?

File: scripts/copy-i18n-tags.coffee [hunk 2]
```diff
@@ -49,10 +51,12 @@ for file in dir when not (file in ['locale.coffee', 'en.coffee', 'rot13.coffee']
       if commentsMap[enCat]? and commentsMap[enCat][enTag]?
         comment = " \##{commentsMap[enCat][enTag]}"
 
+      # fileSource = source code of locale file (ie fr.coffee)
+      # enTag = name of the tranlsation tag (ie matt_title)
```
Comment (AndrewJakubowicz) [inline scripts/copy-i18n-tags.coffee]: Small typo
```suggestion
      # enTag = name of the translation tag (ie matt_title)
```

File: scripts/propagate-i18n-tags.js [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    const localeSource = fs.readFileSync(
+        path.join(__dirname, `../app/locale/${localeFile}`),
+        { encoding: 'utf8'}
+    );
+
+    // Load locale
+    const localeContents = require(`../app/locale/${localeFile}`);
+    const localeTranslations = localeContents.translation || {};
+
+    // Initialie rewrite of file with first line
```
Comment (AndrewJakubowicz) [inline scripts/propagate-i18n-tags.js]: Small typo
```suggestion
    // Initial rewrite of file with first line
```

File: scripts/propagate-i18n-tags.js [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+                    comment += ` ${CHANGE_MARKER}`; // TODO make {change} a constant
+                }
+            }
+
+            comment = comment.trim();
+            if (comment.length > 0) {
+                comment = `# ${comment}`;
+            }
+
+            // If the tag does not exist in the locale file, make sure it is commented otu
```
Comment (AndrewJakubowicz) [inline scripts/propagate-i18n-tags.js]: Typo
```suggestion
            // If the tag does not exist in the locale file, make sure it is commented out
```

File: scripts/propagate-i18n-tags.js [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+            } else {
+                const tagIsMarkedChangeRegex = new RegExp(`^\\s+"?${escapeRegexp(enTagName)}"?:\\s".*"\\s*` +
+                    `#\\s*${escapeRegexp(CHANGE_MARKER)}\\s*$`, 'm');
+
+                // If locale file has tag marked as change and comment is not already marked change,
+                // add change to comment
+                if (localeSource.search(tagIsMarkedChangeRegex) >= 0 &&
+                    comment.search(CHANGE_PATTERN) === -1) {
+
+                    comment += ` ${CHANGE_MARKER}`; // TODO make {change} a constant
```
Comment (AndrewJakubowicz) [inline scripts/propagate-i18n-tags.js]: Is this comment required?

File: scripts/propagate-i18n-tags.js (hunk 1 above)
Comment (jmif) [reply scripts/propagate-i18n-tags.js]: Updated

File: scripts/copy-i18n-tags.coffee (hunk 2 above)
Comment (jmif) [reply scripts/copy-i18n-tags.coffee]: Removed this file

File: scripts/propagate-i18n-tags.js (hunk 3 above)
Comment (jmif) [reply scripts/propagate-i18n-tags.js]: Fixed

File: scripts/propagate-i18n-tags.js (hunk 4 above)
Comment (jmif) [reply scripts/propagate-i18n-tags.js]: Fixed

File: scripts/propagate-i18n-tags.js (hunk 5 above)
Comment (jmif) [reply scripts/propagate-i18n-tags.js]: Nope - removed

**Comment by jmif** [comment]
@shubhi1092 base is updated.  Was pointing to development-environment so that the diff would display properly and I could continue to use the development-environment changes I have during development.  I've rebased those changes out for the final merge.

**Pull request merged**